kong/init.lua

```diff
@@ -629,7 +630,9 @@ function Kong.rewrite()
 
   -- On HTTPS requests, the plugins iterator is already updated in the ssl_certificate phase
   if ngx.var.https ~= "on" then
-    local ok, err = runloop.update_plugins_iterator()
+
+    local timeout = runloop.get_rebuild_timeout()
+    local ok, err = runloop.rebuild_plugins_iterator(timeout)
```

Comment:
I see that you call `runloop.get_rebuild_timeout()` and immediately after `runloop.rebuild_plugins_iterator(timeout)`.  Doesn't `runloop` already know the `timeout`? I am fine with this argument, but when passed as `nil` could `runloop` then just get the timeout on it's own?

So could this be changed to just:
```
local ok, err = runloop.rebuild_plugins_iterator()
```

Reply:
I have replaced this with `runloop.get_updated_plugins_iterator()`, which will:
* get the updated plugins iterator using the proper timeout (now `REBUILD_TIMEOUT`)
* If updating fails, it logs the failure with `CRIT` and just returns the non-updated plugins iterator.
* So this function never returns `nil, err`.

The other function, `runloop.get_plugins_iterator`, still exists and is still being used when we just want to get the current iterator without (maybe) updating it first.